int32 packet_size

int8[] firmware